### ISSUE TITLE
decompose / radical fails in a ring with no degree

### ISSUE THREAD

**Issue by pzinn** [Macaulay2/M2#352]
The example speaks for itself:

i1 : R=QQ[x,DegreeRank=>0]

o1 = R

o1 : PolynomialRing

i2 : decompose ideal x
stdio:2:1:(3): error: array index 0 out of bounds 0 .. -1

**Comment by DanGrayson** [comment]
It works here:
```
Macaulay2, version 1.8.2
with packages: ConwayPolynomials, Elimination, IntegralClosure, LLLBases,
               PrimaryDecomposition, ReesAlgebra, TangentCone

i1 : R=QQ[x,DegreeRank=>0]

o1 = R

o1 : PolynomialRing

i2 :  decompose ideal x

o2 = {ideal x}

o2 : List
```
What version are you using?

**Comment by pzinn** [comment]
ah, oops, sorry. you're right, it seems to work on my computer as well.
I'll investigate, since I do have a more complicated example that fails in
a repeatable manner.

On Thu, Nov 12, 2015 at 3:34 PM, Daniel R. Grayson <notifications@github.com

**Comment by pzinn** [comment]
let me try again:

Macaulay2, version 1.8.2.1
with packages: ConwayPolynomials, Elimination, IntegralClosure, LLLBases,
PrimaryDecomposition, ReesAlgebra, TangentCone

i1 : R=QQ[x,DegreeRank=>0]

o1 = R

o1 : PolynomialRing

i2 : saturate(ideal(x),x)
stdio:2:1:(3): error: array index 0 out of bounds 0 .. -1

**Comment by DanGrayson** [comment]
Okay, I think I've fixed it.  See commit 7997c43ea179d5a7040cf4a3543aba4f68e00390.

Here is the diff:
```
diff --git a/M2/Macaulay2/m2/colon.m2 b/M2/Macaulay2/m2/colon.m2
index 312ae77..484f5c5 100644
--- a/M2/Macaulay2/m2/colon.m2
+++ b/M2/Macaulay2/m2/colon.m2
@@ -291,7 +291,7 @@ saturate(Ideal,Ideal) := Ideal => opts -> (J,I) -> (
     R := ring I;
     if ring J =!= R then error "expected ideals in the same ring";
     if I == 0 then return ideal(1_R);
-    linearvars := (n === 1 and first degree(I_0) === 1 and all((degrees R)/first, x -> x === 1));
+    linearvars := (n === 1 and degreeLength R > 0 and first degree(I_0) === 1 and all((degrees R)/first, x -> x === 1));
     homog := isHomogeneous I and isHomogeneous J;
     
     strategy := opts.Strategy;
@@ -315,7 +315,7 @@ saturate(Ideal,Ideal) := Ideal => opts -> (J,I) -> (
 	)
     else if strategy === null then (
 	 -- if the conditions for Linear hold, we use that algorithm.
-	 if isFlatPolynomialRing R and isGRevLexRing R and homog and linearvars then strategy = Linear
+	 if isFlatPolynomialRing R and homog and linearvars and isGRevLexRing R then strategy = Linear
 	 );
 
     if strategy === Linear then
```

Reopen if there are further problems.
